examples/keyboard_emu/keyboard_emu.cpp

```diff
@@ -0,0 +1,189 @@
+//This example uses pin 40, 38, and 37.
+//GND -> Switch
+//3.3v -> Resistor -> other side of switch + Pin 40/38/37
+//It is assumed pins are held high when the switch is open
+//When a pin goes low from a switch closing, it will be passed as keyboard input.
+//pin40==keyboard_1
+//pin39==keyboard_2
+//pin37==keyboard_3
+
+#include <iostream>
+#include <string>
+#include <cassert>
+#include <stdio.h>
+#include <stdlib.h>
+#include <unistd.h>
+#include <sys/fcntl.h>
+#include <cstring>
+#include <linux/uinput.h>
+
+#include "../../peripheralController/peripheralController.h"
+#include "../../gpioController/gpio.h"
+#include "../../pinmuxController/pinmuxController.h"
+
+//for emitting keyboard events to the OS
+//https://www.kernel.org/doc/html/v4.12/input/uinput.html
+void emit(int fk, int type, int code, int val)
+{
+	struct input_event ie;
+
+	ie.type = type;
+	ie.code = code;
+	ie.value = val;
+	/* timestamp values below are ignored */
+	ie.time.tv_sec = 0;
+	ie.time.tv_usec = 0;
+
+	write(fk, &ie, sizeof(ie));
+}
+
+int main()
+{
+	PeripheralController myGpioController1(gpioController::gpioController1BaseAddress);
```

Comment:
The style I use is indent of 4 spaces for every scope. Can you change it to that?